instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -125,6 +126,26 @@ def process_request(self, request):
         request.META[self._environ_span_key] = span
         request.META[self._environ_token] = token
 
+    def process_view(self, request, view_func, *args, **kwargs):
+        # Process view is executed before the view function, here we get the
+        # route template from request.resolver_match.  It is not set yet in process_request
+        if self._excluded_urls.url_disabled(request.build_absolute_uri("?")):
+            return
+
+        if (
+            self._environ_activation_key in request.META.keys()
+            and self._environ_span_key in request.META.keys()
+        ):
+            span = request.META[self._environ_span_key]
+
+            if span.is_recording():
+                if getattr(request, "resolver_match") and getattr(
```

Comment:
I used getattr because I also don't want to capture the value if it is None.

Reply:
```python

```python
match = getattr(request, "resolved_match")
if match:
    route = getattr(match, "route")
    if route:
        set_attribute()
```

Perhaps use this? I know it's usually not a big deal but we never know where the library will be used and better to reduce attribute look ups if we can in case the code ends up on a very hot path. 
```

Reply:
makes sense...  I have not thought about reducing attribute lookups before.  Thanks!